addons/sale_project/models/product.py

```diff
@@ -28,31 +37,71 @@ class ProductTemplate(models.Model):
-                )
+            if record.service_policy == 'delivered_milestones':
+                if record.service_tracking == 'no':
+                    record.product_tooltip = _(
+                        "Invoice your milestones when they are reached."
+                    )
+                elif record.service_tracking == 'task_global_project':
+                    record.product_tooltip = _(
+                        "Invoice your milestones when they are reached. "
+                        "Create a task in an existing project to track the time spent."
+                    )
+                elif record.service_tracking == 'task_in_project':
+                    record.product_tooltip = _(
+                        "Invoice your milestones when they are reached. "
+                        "Create an empty project for the order to track the time spent."
+                    )
+                elif record.service_tracking == 'project_only':
+                    record.product_tooltip = _(
+                        "Invoice your milestones when they are reached. "
+                        "Create a project for the order with a task for each sales order line "
+                        "to track the time spent."
+                    )
+            elif record.service_policy == 'delivered_manual':
+                if record.service_tracking == 'no':
+                    record.product_tooltip = _(
+                        "Invoice this service when it is delivered (set the quantity by hand on your sales order lines). "
+                    )
+                elif record.service_tracking == 'task_global_project':
+                    record.product_tooltip = _(
+                        "Invoice this service when it is delivered (set the quantity by hand on your sales order lines). "
+                        "Create a task in an existing project to track the time spent."
+                    )
+                elif record.service_tracking == 'task_in_project':
+                    record.product_tooltip = _(
+                        "Invoice this service when it is delivered (set the quantity by hand on your sales order lines). "
+                        "Create an empty project for the order to track the time spent."
+                    )
+                elif record.service_tracking == 'project_only':
+                    record.product_tooltip = _(
+                        "Invoice this service when it is delivered (set the quantity by hand on your sales order lines). "
+                        "Create a project for the order with a task for each sales order line to track the time spent."
+                    )
+
+    @api.onchange('service_policy')
+    def _inverse_service_policy(self):
+        for product in self:
+            if product.service_policy:
+                product.invoice_policy, product.service_type = SERVICE_TO_GENERAL.get(product.service_policy, (False, False))
```

Comment:
To check to be sure, the `SERVICE_TO_GENERAL` has the update when `sale_timesheet` or it just has the content in this file?

Reply:
Yes, it gets updated here:

https://github.com/odoo/odoo/blob/e434a4e4261c0482c1a8a4bc3007697eb79936f7/addons/sale_timesheet/models/product.py#L9-L17

Reply:
Yes, it gets updated

Reply:
Nice, thanks. :)